MoEngagePluginCards/DataTransfer/Sync/MoEngageCardSyncManager.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  MoEngageCardSyncManager.swift
+//  MoEngagePluginCards
+//
+//  Created by Soumya Mahunt on 20/06/23.
+//
+
+import MoEngageCore
+import MoEngageCards
+
+protocol MoEngageCardSyncManagerProtocol {
+    func attachDelegate(_ delegate: MoEngageCardSyncDelegate)
+    func detachDelegate()
+
+    func sendUpdate(
+        forEventType eventType: MoEngageCardsSyncEventType,
+        andAppID appId: String,
+        withNewData data: MoEngageCardSyncCompleteData?
+    )
+}
+
+final class MoEngageCardSyncManager: MoEngageCardSyncManagerProtocol {
+    private let queue: MoEngageCardSyncManagerSynchronizer
+    private let dataManager: MoEngageCardSyncDataManagerProtocol
+    private var delegate: MoEngageCardSyncDelegate?
+    private var syncedOnce: Bool
```

Comment:
@msoumya-engg-sdk why this bool?

Reply:
This is kept to handle scenarios where SDK might call card sync APIs before hybrid gets the chance to call the sync APIs. We can check whether hybrid has already synced based on this flag. What are your thoughts?